homeassistant/exceptions.py

```diff
@@ -69,10 +69,11 @@ def __str__(self) -> str:
             self._message = (
                 f"Parent class {self.__class__.__name__} is missing __str__ method"
             )
-            # If the there is an other super class involved,
-            # we want to call its __str__ method.
-            # If the super().__str__ method is missing in the base_class
-            # the call will be recursive and we return our initialized default.
+            # If a sub class inherits another super class, like ExceptionGroup,
+            # we want to call its __str__ method. If the __str__ method is missing
+            # in the sub class, the call to super().__str__() will be recursive
```

Comment:
Why will the call to super be recursive?

Reply:
Not sure, it just seems it is, and the tests prove that.

Reply:
The `ExceptionGroup` base class seems to implement `__str__` but `Exception` dos not.

Reply:
> The `ExceptionGroup` base class seems to implement `__str__` but `Exception` dos not. In case of  as sub class implementing a constructor. but no `__str__`. `super().__str__` seems to default to the `__str__` method of `HomeAssistantError`. That might cause a recursion loop.

Reply:
```py
Python 3.11.8 (main, Feb 25 2024, 16:41:26) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: test = Exception("Boom!")

In [2]: test.__str__()
Out[2]: 'Boom!'
```

Reply:
I think it is what `super().__str__` str is referencing is `HomeAssistantError.__str__` in case of a sub class with only inherriting `HomeAssistantError`.